### DISCUSSION TITLE
How do I make sure that all http calls from a web page go through the same URL + path?

### DISCUSSION THREAD

**Discussion by Timmiej93** [NginxProxyManager/nginx-proxy-manager#1301] [Q&A]
I feel like this is something that has to be done by the webpage developer, but since I'm not sure, I'll ask:

I got custom locations working, `url.com/serviceX` redirects correctly to the desired location. However, every single http call that comes from that web page after that returns a 404, since it tries to get everything from `url.com` instead of `url.com/serviceX`. For example, the webpage would try to load `url.com/stylesheet.css` instead of `url.com/serviceX/stylesheet.css`. Is this something that can be handled in NPM?

**Comment by chaptergy** [comment]
Unfortunately this would be something which depends heavily on the app you are running, since it generates the invalid URLs.
There are [some options which would fix this](https://serverfault.com/questions/796930/nginx-is-giving-404-errors-on-all-but-the-html-pages/797006#797006). But is not that intuitively doable in npm. Option 3 would be possible through the advanced config tab.

**Reply by Timmiej93** [reply]
Are you sure option 3 is possible? It uses the `$http_referer` and `$request_uri` variables, NPM seems to only give you `$server`, `$port` and `$forward_scheme`. I tried setting a test header with the value of `$http_referer`, but it doesn't seem to contain anything.

**Reply by chaptergy** [reply]
The two variables are provided by nginx, so they are available. When I add the following advanced config to a custom location I can see that the variables are set.

```nginx
add_header X-referrer $http_referer;
add_header X-uri $request_uri;
```

They are not set for the inital request of a page, e.g. `index.html`, but should be set when requesting all assets contained in this html page.
